class/killboardcheck.class.php

```diff
@@ -0,0 +1,75 @@
+<?php
+// use database class
+
+// check if called from an allowed page
+if (!defined('ESRC'))
+{
+	echo "Do not call the script directly!";
+	exit ( 1 );
+}
+
+require_once '../class/db.class.php';
+
+/**
+ * Class to check killboard details. 
+ *
+ * CREATE TABLE `esrcuser`.`killboard_check` ( `lastUpdated` DATETIME NULL ) ENGINE = InnoDB;
+ *
+ */
+class KillboardActivity {
+	
+	var $db = null;
+	
+	public function __construct($database = NULL)
+	{
+		if (isset($database))
+		{
+			$this->db = $database;
+		}
+		else
+		{
+			// create a new database class instance
+			$this->connectDatabase ();
+		}
+	}
+	
+	/**
+	 * Creates a new DB connection.
+	 */
+	private function connectDatabase() {
+		$this->db = new Database ();
+	}
+		
+	/**
+	 * Get last updated timestamp to see when we checked the killboard
+	 * @return timestamp
+	 */
+	public function getLastKillboardCheck()
+	{
+		$selectStmt = "SELECT lastUpdated FROM killboard_check;";
```

Comment:
Is the 'killboard_check' a new table with just one entry? What is the table definition. Should we think about a system status table like 'create table systemstatus(name varchar..., updated timestamp, maybeavalue ...)' and can handle different states? Not sure if this is necessary/helpful at different places. I'm fune with the simple table for the moment. But the definition is missing. But should support multiple rows with 'select max(lastupdated) from ...' because can be executed in parallel.